test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs

```diff
@@ -141,15 +334,327 @@ public void CompareKeyInfoTest()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareSignatures", true);
+            var context = new CompareContext($"{this}.CompareSignatures");
+            var signature1 = new Signature {SignatureValue = Guid.NewGuid().ToString() };
+            var signature2 = new Signature {SignatureValue = Guid.NewGuid().ToString()};
+            IdentityComparer.AreEqual(signature1, signature2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "SignatureValue:") == 1);
+        }
+
+        [Fact]
+        public void CompareSignedInfo()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareSignedInfo", true);
+            var context = new CompareContext($"{this}.CompareSignedInfo");
+            var signedInfo1 = new SignedInfo { SignatureMethod = Guid.NewGuid().ToString() };
+            var signedInfo2 = new SignedInfo { SignatureMethod = Guid.NewGuid().ToString() };
+            IdentityComparer.AreEqual(signedInfo1, signedInfo2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "SignatureMethod:") == 1);
+        }
+
+        [Fact]
+        public void CompareSigningCredentials()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareSigningCredentials", true);
+            var context = new CompareContext($"{this}.CompareSigningCredentials");
+            var signingCredentials1 = new SigningCredentials(KeyingMaterial.DefaultX509Key_2048, Guid.NewGuid().ToString());
+            var signingCredentials2 = new SigningCredentials(KeyingMaterial.DefaultX509Key_2048, Guid.NewGuid().ToString());
+            IdentityComparer.AreEqual(signingCredentials1, signingCredentials2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "Algorithm:") == 1);
+        }
+
+        [Fact]
+        public void CompareStrings()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareStrings", true);
+            var context = new CompareContext($"{this}.CompareStrings");
+            var string1 = "hello";
+            var string2 = "goodbye";
+            IdentityComparer.AreEqual(string1, string2, context);
+
+            Assert.True(context.Diffs.Count(s => s == $"'{string1}'") == 1);
+            Assert.True(context.Diffs.Count(s => s == "!=") == 1);
+            Assert.True(context.Diffs.Count(s => s == $"'{string2}'") == 1);
+        }
+
+        [Fact]
+        public void CompareSymmetricSecurityKeys()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareSymmetricSecurityKeys", true);
+            var context = new CompareContext($"{this}.CompareSymmetricSecurityKeys");
+            var symmetricSecurityKey1 = new SymmetricSecurityKey(KeyingMaterial.SymmetricKeyBytes2_256);
+            var symmetricSecurityKey2 = new SymmetricSecurityKey(KeyingMaterial.SymmetricKeyBytes2_128);
+            IdentityComparer.AreEqual(symmetricSecurityKey1, symmetricSecurityKey2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "KeySize:") == 1);
```

Comment:
byte[] is not a type that was located in the _equalityDict, so the two byte arrays were never actually compared. I modified the AreBytesEqual method in the IdentityComparer so that it would actually find this difference.